packages/aws-client/src/S3.ts

```diff
@@ -393,18 +414,45 @@ export const putJsonS3Object = (bucket: string, key: string, data: any) =>
     Body: JSON.stringify(data)
   });
 
+/**
+ * Get a readable stream for an S3 object
+ *
+ * @param {Object} params
+ * @param {AWS.S3} params.s3 - an AWS.S3 instance
+ * @param {string} params.bucket - the bucket of the requested object
+ * @param {string} params.key - the key of the requested object
+ * @returns {Readable}
+ */
+export const getObjectReadStream = (params: {
+  s3: { getObject: GetObjectMethod },
+  bucket: string,
+  key: string
+}) => {
+  // eslint-disable-next-line no-shadow
+  const { s3, bucket, key } = params;
+
+  return s3.getObject({ Bucket: bucket, Key: key }).createReadStream();
+};
+
 /**
  * Get a readable stream for an S3 object.
  *
  * @param {string} bucket - the S3 object's bucket
  * @param {string} key - the S3 object's key
  * @returns {ReadableStream}
  * @throws {Error} if S3 object cannot be found
+ *
+ * @deprecated
  */
-export const getS3ObjectReadStream = (bucket: string, key: string) =>
-  s3().getObject(
-    { Bucket: bucket, Key: key }
-  ).createReadStream();
+export const getS3ObjectReadStream = (bucket: string, key: string) => {
+  deprecate(
+    '@cumulus/aws-client/S3.getS3ObjectReadStream',
+    '1.24.0',
+    '@cumulus/aws-client/S3.getObjectReadStream'
+  );
+
+  return getObjectReadStream({ s3: s3(), bucket, key });
+};
```

Comment:
If using `util.deprecate()` and `deprecationMessage()`, this would become:

```javascript
export const getS3ObjectReadStream = util.deprecate(
  (bucket: string, key: string) => getObjectReadStream({ s3: s3(), bucket, key }),
  deprecationMessage(
    '@cumulus/aws-client/S3.getS3ObjectReadStream',
    '1.24.0',
    '@cumulus/aws-client/S3.getObjectReadStream'
  )
);

Reply:
The other complication here is that our log messages are supposed to be in a JSON format, which `util.deprecate` does not support out of the box. I added a `buildMessage` method to the Logger class that will return a properly formatted message, which can then be used with `util.deprecate`.

Reply:
I don't quite follow the log message stuff, but no worries.